### PR TITLE
fix: remove 'Fake' funding strategy

### PR DESCRIPTION
**Pull request by snario**
The `"Fake"` funding strategy was initially added to simplify development of the server wallet. It has lived out its initial purpose, and now must be terminated. Fake funding functionality is provided by using the `"Direct"` funding type and adding extra calls to the `Wallet`'s `updateChannelFunding` with fake data.

### CODE REVIEW

**Comment by snario** [comment]
This is an alternative to #3154

**Comment by andrewgordstewart** [comment]
Before merging this, don't we want to be certain we can implement the required changes to the `MockChainService` to
> [add] extra calls to the Wallet's updateChannelFunding with fake data.
?

**Comment by snario** [comment]
Yeah. Working on this now.

File: packages/server-wallet/src/protocols/__test__/defund-channel.test.ts [hunk 1]
```diff
@@ -187,26 +187,6 @@ it('should fail the objective when the channel does not exist', async () => {
   expect(reloadedObjective.status).toEqual('failed');
 });
 
-it('should fail when using non-direct funding', async () => {
-  const chainService = new MockChainService();
-  const channelDefunder = ChannelDefunder.create(store, chainService, logger, timingMetrics);
-
-  // Create a channel with fake funding strategy
-  const c = channel({fundingStrategy: 'Fake'});
```
Comment (lalexgap) [inline packages/server-wallet/src/protocols/__test__/defund-channel.test.ts:195]: I think we should keep this test but use ledger funding:
 ` const c = channel({fundingStrategy: 'Ledger'});`

**Review by lalexgap — DISMISSED** [review]
LGTM

**Comment by lalexgap** [comment]
> LGTM

Ha nevermind, didn't notice the `do-not-merge` label

**Comment by snario** [comment]
Removing fake funding makes it difficult to co-ordinate multiple wallets across different processes since they would need to rely on a fake `ChainService` emitting events based on the behaviour of a counterparty which is not yet implemented. For now we'll keep Fake funding until a better alternative comes along.

**Pull request closed without merging**